schema/training-module.js

```diff
@@ -12,10 +12,10 @@ class TrainingModule extends BaseModel {
       required: ['profile_id'],
       additionalProperties: false,
       properties: {
-        id: { type: 'string', pattern: uuid.v4 },
+        id: { type: ['string', 'null'], pattern: uuid.v4 },
```

Comment:
I don't think this _should_ be nullable. There's no context in which it's ok to unset the id.